plugins/analytics-module-ga4/src/apis/implementations/AnalyticsApi/GoogleAnalytics4.test.ts

```diff
@@ -0,0 +1,419 @@
+      api.captureEvent({
+        action: expectedAction,
+        subject: expectedLabel,
+        value: expectedValue,
+        context,
+      });
+      expect(fnSend).toHaveBeenCalledWith({
+        hitType: 'event',
+        eventAction: 'search',
+        eventCategory: 'App',
+        eventLabel: 'search-term',
+        eventValue: 42,
+        search_term: 'search-term',
+      });
+    });
+
+    it('tracks basic event', () => {
+      const api = GoogleAnalytics4.fromConfig(basicValidConfig);
+
+      const expectedAction = 'click';
+      const expectedLabel = 'on something';
+      const expectedValue = 42;
+      api.captureEvent({
+        action: expectedAction,
+        subject: expectedLabel,
+        value: expectedValue,
+        context,
+      });
+
+      expect(fnSend).toHaveBeenCalledWith({
+        hitType: 'event',
+        eventCategory: context.extension,
+        eventAction: expectedAction,
+        eventLabel: expectedLabel,
+        eventValue: expectedValue,
+      });
+    });
+
+    it('captures configured custom dimensions/metrics on pageviews', () => {
+      const api = GoogleAnalytics4.fromConfig(advancedConfig);
+      api.captureEvent({
+        action: 'navigate',
+        subject: '/a-page',
+        context,
+      });
+
+      expect(fnSend).toHaveBeenCalledWith({
+        hitType: 'pageview',
+        page: '/a-page',
+        c_pluginId: context.pluginId,
+        c_releaseNum: context.releaseNum,
+      });
+    });
+
+    it('captures configured custom dimensions/metrics on events', () => {
+      const api = GoogleAnalytics4.fromConfig(advancedConfig);
+
+      const expectedAction = 'search';
+      const expectedLabel = 'some query';
+      const expectedValue = 5;
+      api.captureEvent({
+        action: expectedAction,
+        subject: expectedLabel,
+        value: expectedValue,
+        attributes: {
+          extraDimension: false,
+          extraMetric: 0,
+        },
+        context,
+      });
+
+      expect(fnSend).toHaveBeenCalledWith({
+        hitType: 'event',
+        eventCategory: context.extension,
+        eventAction: expectedAction,
+        eventLabel: expectedLabel,
+        eventValue: expectedValue,
+        c_pluginId: context.pluginId,
+        c_releaseNum: context.releaseNum,
+        search_term: expectedLabel,
+      });
+    });
+
+    it('does not pass non-numeric data on metrics', () => {
+      const api = GoogleAnalytics4.fromConfig(advancedConfig);
+
+      api.captureEvent({
+        action: 'verb',
+        subject: 'noun',
+        attributes: {
+          extraMetric: 'not a number',
+        },
+        context,
+      });
+
+      expect(fnEvent).not.toHaveBeenCalledWith({
+        eventCategory: context.extension,
+        eventAction: 'verb',
+        eventLabel: 'noun',
+        c_pluginId: context.pluginId,
+        c_releaseNum: context.releaseNum,
+        c_extraMetric: 'not a number',
+      });
+    });
+  });
```

Comment:
Now that dimensions/metrics are just captured as "parameters" this test is probably no longer relevant, eh?